app/src/main/java/com/hedvig/app/feature/profile/ui/payment/PaymentViewModel.kt

```diff
@@ -17,34 +21,36 @@ import kotlinx.coroutines.launch
 abstract class PaymentViewModel(
     hAnalytics: HAnalytics,
 ) : ViewModel() {
-    protected val _paymentData = MutableLiveData<PaymentQuery.Data>()
-    protected val _payinStatusData = MutableLiveData<PayinStatusQuery.Data>()
-    val data = combineTuple(_paymentData, _payinStatusData)
+    protected val _paymentData = MutableStateFlow<PaymentQuery.Data?>(null)
+    protected val _payinStatusData = MutableStateFlow<Pair<PayinStatusQuery.Data?, PaymentType>?>(null)
+    val data: StateFlow<Pair<PaymentQuery.Data?, Pair<PayinStatusQuery.Data?, PaymentType>?>> =
+        combineState(_paymentData, _payinStatusData, viewModelScope)
```

Comment:
combineState turns everything into nullables right. Can we not use this instead https://github.com/Kotlin/kotlinx.coroutines/blob/52cbf01c63c0737ee745ff08568f26385271a9ab/kotlinx-coroutines-core/common/src/flow/operators/Zip.kt#L51-L52 and handle the null cases on the ViewModel with a local `ViewState` (sealed?) object instead of in the call site? Losing the nullability information and making it always nullable doesn't sound that much fun.

Something like this https://github.com/HedvigInsurance/android/blob/48e4e9f44ec5484ed830b0b259e6b0f468b00bc7/app/src/main/java/com/hedvig/app/feature/addressautocompletion/ui/AddressAutoCompleteViewModel.kt#L82-L95 where all the inputs are combined into a ViewState.

Reply:
I was mostly just trying to mimic exactly the utility that was in place before (`combineTuple`), but I think I agree - being explicit is likely better here.